src/charm.py

```diff
@@ -256,7 +256,10 @@ def _is_valid_timespec(self, timeval) -> bool:
         Returns:
             True if time specification is valid and False otherwise.
         """
-        if not (matched := re.match(r"[1-9][0-9]*[ymwdhs]", timeval)):
+        timespec_re = re.compile(
```

Comment:
Add backlink to the official docs?

Reply:
Yeah a comment about a regex like this is definitely needed.

Reply:
Linking to the source is great.
I was thinking about the [doc page](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#configuration-file) but this is probably better, at least for now.